cpp/src/aggregation/aggregation.cpp

```diff
@@ -22,142 +22,231 @@
 
 namespace cudf {
 
-std::vector<aggregation::Kind> aggregation::get_simple_aggregations(data_type col_type) const
+std::vector<std::unique_ptr<aggregation>> aggregation::get_simple_aggregations(
+  data_type col_type) const
 {
-  return {this->kind};
-}
-void aggregation::finalize(cudf::detail::aggregation_finalizer& finalizer)
-{
-  finalizer.visit(*this);
+  std::vector<std::unique_ptr<aggregation>> aggs;
+  aggs.push_back(clone());
+  return aggs;
```

Comment:
```suggestion
  return {clone()};
```

Reply:
Can't construct a vector of unique_ptr<>s like this.

```
projects/cudf/cpp/src/aggregation/aggregation.cpp:31:18:   required from here
/usr/include/c++/9/bits/stl_uninitialized.h:127:72: error: static assertion failed: result type must be constructible from value type of input range
  127 |       static_assert(is_constructible<_ValueType2, decltype(*__first)>::value,
```